staging/src/k8s.io/apiserver/pkg/util/globalflag/globalflags.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package globalflag
```

Comment:
I would prefer not to add this to apiserver. There's no reason for scheduler to depend on apiserver afaik and I think it should stay that way?

Reply:
How about  put `globalflags.go` in `pkg/util/flag` package? seems all `component` in `cmd/` can reference it

Reply:
Fixed , PTAL. Thanks

Reply:
> I would prefer not to add this to apiserver. There's no reason for scheduler to depend on apiserver afaik and I think it should stay that way?

@lavalamp all those components depend on apiserver already as they serve APIs (though no API groups of course, but /healthz, /metrics, /debug) and make use of apiserver code like delegated authn/z, SNI, etc.

Reply:
@sttts :/ I wonder if those parts should go to their own packages.

Reply:
@lavalamp We had recent discussions about a `k8s.io/component` middle ground, somewhat below `k8s.io/apiserver`. Currently kube component-wide packages either go to apimachinery or apiserver, without dependencies to utils. Component configs could also benefit from k8s.io/component.

Reply:
Let's move forward with this PR for now. I think in the light of the current role of `k8s.io/apiserver` to be the root for all kube components (let's change that in 1.14!) the globalflags package is fine.

Reply:
apiserver for now to be moved to a k8s.io/component package later SGTM, thanks.